client/lib/protect-form/index.tsx

```diff
@@ -62,60 +84,31 @@ export interface ProtectedFormProps {
  */
 export const protectForm = < P extends ProtectedFormProps >(
```

Comment:
We could use `createHigherOrderComponent` and its typings here.

Reply:
https://github.com/Automattic/wp-calypso/pull/52054/commits/f185cdf02c2350415306a8c13f758ac3781acb15 👍🏻